driver/allclose_driver.hpp

```diff
@@ -0,0 +1,343 @@
+                                            input1_dev->GetMem(),
+                                            input2Desc,
+                                            input2_dev->GetMem(),
+                                            outputDesc,
+                                            output_dev->GetMem(),
+                                            atol,
+                                            rtol,
+                                            equal_nan,
+                                            workspace_dev->GetMem(),
+                                            ws_sizeInBytes);
+
+        MIOPEN_THROW_IF(status != miopenStatusSuccess, "Error in miopenAllCloseForward");
+
+        float time = 0.0;
+        miopenGetKernelTime(GetHandle(), &time);
+        kernel_total_time += time;
+        if(i == 0)
+            kernel_first_time = time;
+    }
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        STOP_TIME
+        int iter = inflags.GetValueInt("iter");
+        if(WALL_CLOCK)
+            std::cout << "Wall-clock Time Forward AllClose Elapsed: " << t.gettime_ms() / iter
+                      << " ms" << std::endl;
+
+        float kernel_average_time =
+            iter > 1 ? (kernel_total_time - kernel_first_time) / (iter - 1) : kernel_first_time;
+        std::cout << "GPU Kernel Time Forward AllClose Elapsed: " << kernel_average_time << " ms"
+                  << std::endl;
+    }
+
+    if(output_dev->FromGPU(GetStream(), output.data()) != 0)
+    {
+        std::cerr << "Error copying (output_dev) from GPU, size: " << output_dev->GetSize()
+                  << std::endl;
+        return miopenStatusInternalError;
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tout>
+int AllCloseDriver<Tgpu, Tout>::RunForwardCPU()
+{
+    int status = miopenStatusSuccess;
+
+    status = mloAllCloseForwardRunHost<Tgpu, Tout>(input1Desc,
+                                                   input1.data(),
+                                                   input2Desc,
+                                                   input2.data(),
+                                                   output_host.data(),
+                                                   atol,
+                                                   rtol,
+                                                   equal_nan);
+    MIOPEN_THROW_IF(status != miopenStatusSuccess, "Error in mloAllCloseForwardRunHost");
+
+    return status;
+}
+
+template <typename Tgpu, typename Tout>
+int AllCloseDriver<Tgpu, Tout>::RunBackwardGPU()
+{
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tout>
+int AllCloseDriver<Tgpu, Tout>::RunBackwardCPU()
+{
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tout>
+int AllCloseDriver<Tgpu, Tout>::VerifyForward()
+{
+    RunForwardCPU();
+
+    auto error = miopen::rms_range(output_host, output);
+    if(!std::isfinite(error) || output_host[0] != output[0])
```

Comment:
operation `==` already includes length check between 2 vectors
```suggestion
    if(output_host.data == output.data)
```

Reply:
In other PR, this should be `if(!std::isfinite(error) || error > tolerance)` . May I ask the reason why you implemented like above?

Reply:
This implementation also check for inf or nan that can occur to both output_host and output so I think we should leave it as is